USBInterface/DeviceScanner.cs

```diff
@@ -41,9 +63,21 @@ public DeviceScanner(ushort vendorId, ushort productId, int scanIntervalMillisec
         }
 
         // scanning for device when it is open by another process will return false
-        public static bool ScanOnce(ushort vid, ushort pid)
+        public static List<HidDeviceInfo> ScanOnce(ushort vid, ushort pid)
         {
-            return HidApi.hid_enumerate(vid, pid) != IntPtr.Zero;
+            var list = new List<HidDeviceInfo>();
+
+            var pDev = HidApi.hid_enumerate(vid, pid);
+            while (pDev != IntPtr.Zero)
+            {
+                var dev = (HidDeviceInfo)Marshal.PtrToStructure(pDev, typeof(HidDeviceInfo));
+                list.Add(dev);
+                // freeing the enumeration releases the device, 
+                // do it as soon as you can, so we dont block device from others
+                HidApi.hid_free_enumeration(pDev);
```

Comment:
`hid_free_enumeration()` frees the entire list and should be called only once after the loop finished (with pointer returned by `hid_enumerate()` as parameter).